### ISSUE TITLE
grpc streaming connection breaks

### ISSUE THREAD

**Issue by arunkjn** [nghttp2/nghttp2#1017]
Hi,

I was able to use this nghttp2 as an ingress controller via nghttpx for my grpc service in kubernetes. It handles tls termination correctly but my rpc subscription which is an indefinite stream disconnect after sometime with the following message on the client-
```
WARNING: RPC failed: Status{code=INTERNAL, description=Connection closed with unknown cause, cause=null}
```
Any idea what might be causing this? Do I need to configure something?

**Comment by tatsuhiro-t** [comment]
nghttpx has a timeout to close idle backend connection.  It is 1 minute by default. Does it apply to your situation?

**Comment by arunkjn** [comment]
Yes I think so. Can you please point me to the relevant config docs? Also is there a way to keep it open indefinitely?

**Comment by arunkjn** [comment]
is this what i am looking for? https://nghttp2.org/documentation/nghttpx.1.html#cmdoption-nghttpx--stream-write-timeout

**Comment by tatsuhiro-t** [comment]
I think the read timer got triggered.  If I'm correct, the relevant options are:

* https://nghttp2.org/documentation/nghttpx.1.html#cmdoption-nghttpx--backend-read-timeout

**Comment by arunkjn** [comment]
@tatsuhiro-t I think that I can succesfully connect to the backend which open an http2 stream connection. This stream connection breaks after sometime. Do you still think that read timeout is responsible for this?

**Comment by arunkjn** [comment]
Also, on a sidenote, if i want to log errors as well in nghttp2x in the same stream as stdout what should i do?

**Comment by tatsuhiro-t** [comment]
Do you mean that --backend-read-timeout does not work for you?
There are several reasons why stream or connection is closed.  Timeout is one of them.
